### ISSUE TITLE
add_class() doesn't preserve existing classes

### ISSUE THREAD

**Issue by cpsievert** [wch/staticimports#2]
```r
add_class(structure(1, class = "foo"), "bar")
[1] 1
attr(,"class")
[1] "bar"
```

**Comment by wch** [comment]
This is a little tricky because if you do `class(x) <- c(class,class(x))`, it will add implicit class names (like `"list"`):

```R
add_class <- function(x, class) {
  class(x) <- c(class,class(x))
  x
}

a <- list()
add_class(a, "foo")
#> list()
#> attr(,"class")
#> [1] "foo"  "list"
```

Contrast this with the usual ways of adding class:

```R
b <- structure(list(), class = "foo")
b
#> list()
#> attr(,"class")
#> [1] "foo"

d <- list()
class(d) <- "foo"
d
#> list()
#> attr(,"class")
#> [1] "foo"
```

**Comment by wch** [comment]
Closed by cde4fbe564b83688bc5900a3124808d192584dbe.

**Comment by cpsievert** [comment]
Yea, `set_class()` is better. This is what I've used in some projects to add/prefix a class:

```r
prefix_class <- function (x, y) {
  # Can't set attributes on a symbol, but that's alright because
  # we don't need to flag or compute on symbols
  if (is.symbol(x)) return(x)
  oldClass(x) <- unique(c(y, oldClass(x)))
  x
}
```
